dashboard/app/views/levels/editors/_applab.html.haml

```diff
@@ -70,7 +70,7 @@
           { "entries": ["function1", "function2"], "matchType": "exact", "minTimes": 1, "message": "You need to call both functions." },
           { "entries": ["[forTest]", "moveTo", "randomNumber", "randomNumber"], "matchType": "inexact", "minTimes": 20, "message": "You need to call moveTo in a for loop." }
         ]
-  = f.text_area 'log_conditions', placeholder: 'Insert JSON Data', rows: 4, value: @level.log_conditions ? JSON.pretty_generate(@level.log_conditions) : ''
+  = f.text_area 'log_conditions', placeholder: 'Insert JSON Data', rows: 4, value: @level.log_conditions or ''
```

Comment:
Any reason not to use `||` rather than `or`?

Not understanding the context, does it make sense to try and `pretty_generate`, keep it if it is error-free, and otherwise fallback to the empty string?